s4e-backend/src/main/java/pl/cyfronet/s4e/controller/InvitationController.java

```diff
@@ -40,13 +55,36 @@
             @ApiResponse(responseCode = "404", description = "Not found", content = @Content)
     })
     @PostMapping(value = "/institutions/{institution}/invitations", consumes = APPLICATION_JSON_VALUE)
-    public void send(
-            @RequestBody @Valid CreateInvitationRequest request,
+    public BasicInvitationResponse send(
+            @RequestBody @Valid InvitationRequest request,
             @PathVariable("institution") String institutionSlug
     ) throws Exception {
         val token = invitationService.createInvitationFrom(request.getEmail(), institutionSlug);
         val invitationEvent = new OnSendInvitationEvent(token, LocaleContextHolder.getLocale());
         eventPublisher.publishEvent(invitationEvent);
+
+        return invitationService.findByToken(token, BasicInvitationResponse.class).get();
+    }
+
+    @Operation(summary = "Resend invitation and delete last one")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "Invitation was resend"),
+            @ApiResponse(responseCode = "400", description = "Incorrect request", content = @Content),
+            @ApiResponse(responseCode = "401", description = "Unauthenticated", content = @Content),
+            @ApiResponse(responseCode = "403", description = "Forbidden", content = @Content),
+            @ApiResponse(responseCode = "404", description = "Not found", content = @Content)
+    })
+    @PutMapping(value = "/institutions/{institution}/invitation/{token}", consumes = APPLICATION_JSON_VALUE)
+    public void resend(
```

Comment:
Why not return BasicInvitationResponse here? It would make sense for me.
Also, as you're creating and deleting in a single method, I'd annotate it with `@Transactional`, or have a single method in a service which does the two with that annotation.